static/src/javascripts/projects/common/modules/identity/okta.ts

```diff
@@ -0,0 +1,65 @@
+import type { CustomClaims, IdentityAuthState } from '@guardian/identity-auth';
+import { IdentityAuth } from '@guardian/identity-auth';
+
+// the `id_token.profile.theguardian` scope is used to get custom claims
+export type CustomIdTokenClaims = CustomClaims & {
+	email: string;
+	braze_uuid: string;
+};
+
+const getStage = () => {
+	return window.guardian.config.page.stage;
+};
+
+const getIssuer = (stage: Stage) =>
+	stage === 'PROD'
+		? 'https://profile.theguardian.com/oauth2/aus3xgj525jYQRowl417'
+		: 'https://profile.code.dev-theguardian.com/oauth2/aus3v9gla95Toj0EE0x7';
+
+const getClientId = (stage: Stage) =>
+	stage === 'PROD' ? '0oa79m1fmgzrtaHc1417' : '0oa53x6k5wGYXOGzm0x7';
+
+const getRedirectUri = (stage: Stage) => {
+	switch (stage) {
+		case 'PROD':
+			return 'https://www.theguardian.com/';
+		case 'CODE':
+			return 'https://m.code.dev-theguardian.com/';
+		case 'DEV':
+		default:
+			return 'http://localhost:9000/';
+	}
+};
+
+let identityAuth: IdentityAuth<never, CustomIdTokenClaims> | undefined =
+	undefined;
```

Comment:
tiny thing: you don't need to specify `= undefined` you could just have

```suggestion
let identityAuth: IdentityAuth<never, CustomIdTokenClaims> | undefined;
```